cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -109,13 +111,68 @@ class column_wrapper {
  *`[begin,end)`
  **/
 template <typename Element, typename InputIterator>
-rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
+rmm::device_buffer make_elements(
+  InputIterator begin,
+  InputIterator end,
+  typename std::enable_if<
+    std::is_convertible<typename std::iterator_traits<InputIterator>::value_type,
+                        Element>::value>::type* = nullptr)
 {
   static_assert(cudf::is_fixed_width<Element>(), "Unexpected non-fixed width type.");
   thrust::host_vector<Element> elements(begin, end);
   return rmm::device_buffer{elements.data(), elements.size() * sizeof(Element)};
 }
 
+// This type trait can be specialized for different `From` and/or `To` types, or the conditional
+// within it can be nested for future types. This is used in the context of `make_elements`
+// when `Element` cannot be implicitly constructed from the items returned by the `InputIterator`.
+template <typename From, typename To>
+struct argument_type_for_explicit_construction {
+  using argument_type = typename std::conditional<
```

Comment:
Instead of nesting `std::conditional`, it would be more canonical to SFINAE the trait for `is_duration_t<To>`.  Nested `std::conditional`s are hard to read.

Reply:
not applicable, as the tests have changed to pass the appropriate types.